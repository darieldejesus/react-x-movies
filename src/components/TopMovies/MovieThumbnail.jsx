import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';

const Container = Styled.div`
  display: block;
  width: 154px;
`;

const Image = Styled.img`
  background: red;
  height: 231px;
  width: 154px;
`;

const MovieName = Styled.a`
  color #63676f;
  font-size: 1rem;
  font-weight: 300;
  margin: 10px 0 0;
  text-decoration: none;
  text-transform: uppercase;
`;

const MovieReleaseDate = Styled.h4`
  color: #ed0005;
  font-size: 0.8rem;
  font-weight: 600;
  margin: 0;
`;

const MovieThumbnail = ({ id, title, poster_path, release_date }) => (
  <Container>
    <Image src={`https://image.tmdb.org/t/p/w154${poster_path}`} />
    <MovieName href={`/movie/${id}`}>{ title }</MovieName>
    <MovieReleaseDate>({ release_date })</MovieReleaseDate>
  </Container>
);

MovieThumbnail.propTypes = {
  id: PropTypes.number.isRequired,
  poster_path: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default MovieThumbnail;
