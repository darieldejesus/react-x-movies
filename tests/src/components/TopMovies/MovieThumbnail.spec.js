import React from 'react';
import MovieThumbnail from '../../../../src/components/TopMovies/MovieThumbnail';
import fixture from '../../../fixture/moviesDetails.json';

describe('MovieThumbnail', function () {
  beforeEach(() => {
    this.wrapper = shallow(<MovieThumbnail {...fixture} />);
  });

  test('should renders ListOfMovies', () => {
    expect(toJson(this.wrapper)).toMatchSnapshot();
  });
});
