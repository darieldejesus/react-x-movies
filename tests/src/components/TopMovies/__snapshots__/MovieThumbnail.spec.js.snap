// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieThumbnail should renders ListOfMovies 1`] = `
<styled.div>
  <styled.img
    src="https://image.tmdb.org/t/p/w154/9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
  />
  <styled.a
    href="/movie/346364"
  >
    It
  </styled.a>
  <styled.h4>
    (
    2017-09-05
    )
  </styled.h4>
</styled.div>
`;
